logger/logger.go

```diff
@@ -140,6 +140,9 @@ func (l *Logger) SetPrefix(prefix string) {
 
 // Debug appends a debug message to the log
 func (l *Logger) Debug(v ...interface{}) {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+
```

Comment:
Query: Do we really need to guard this block? Secondly if it is necessary, we might have to add it for the Info, Warn, Error also.

Reply:
This was added due to some race condition issues that happen when we make changes to the instrumentations.
It doesn't make sense to have this change in this PR, but there is a good chance that I will re-add this change in the coming PR, which is, the one that updates every instrumentation to use the latest core version, where the global sensor is removed from the tests.

Then we can discuss the need to add them for all methods. I would initially argue that we should only lock the code where a race condition actually is present, rather than adding them because a similar method needs it. But I would love to discuss together before deciding on one thing or the other :)